### PR TITLE
Added implementation of LU decomposition

### PR DESCRIPTION
**Pull request by felix642**

### CODE REVIEW

File: src/lu_decomposition.c [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ *
+ * @param initialMatrix The matrix to decompose
+ * @param lMatrix Output matrix L
+ * @param uMatrix Output matrix U
+ * @param size Size of the matrix
+ */
+void LUDecomposition(const lu_real* initialMatrix, lu_real* lMatrix,
+                     lu_real* uMatrix, const int size) {
+  lu_real tmpMatrix[size * size];
+  memcpy(tmpMatrix, initialMatrix, size * size * sizeof(lu_real));
```
Comment (bisson2000) [inline src/lu_decomposition.c:17]: Could you create a variable `sizeSquared = size * size` to reduce multiplications?

**Review by bisson2000 — CHANGES_REQUESTED** [review]
Please remove your `a.out` file.
Also, do we want to add this to the `1chipml.h` header file?

**Review by bisson2000 — APPROVED** [review]
GFM

File: src/lu_decomposition.c [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+/**
+ * @brief Perform the LU decomposition algorithm on the matrix initialMatrix
+ *
+ * @param initialMatrix The matrix to decompose
+ * @param lMatrix Output matrix L
+ * @param uMatrix Output matrix U
+ * @param size Size of the matrix
+ */
+void LUDecomposition(const lu_real* initialMatrix, lu_real* lMatrix,
+                     lu_real* uMatrix, const int size) {
```
Comment (GM07) [inline src/lu_decomposition.c:15]: you made me thought of this (i'm not sure we have been doing it so far), but aren't output parameters usually declared as the last parameters ?

File: src/lu_decomposition.c [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  for (int i = 0; i < size; ++i) {
+    for (int j = 0; j < size - i; ++j) {
+      int col = i + j;
+      uMatrix[coordToIndex(j, col, size)] =
+          tmpMatrix[coordToIndex(j, col, size)];
+    }
+    for (int j = 0; j < i + 1; ++j) {
+      if (i == j) {
+        lMatrix[coordToIndex(i, j, size)] = 1;
+      } else {
```
Comment (GM07) [inline src/lu_decomposition.c:43]: remove this condition and put `lMatrix[coordToIndex(i, i, size)] = 1` above outside of the for loop ?

File: src/lu_decomposition.c (hunk 2 above)
Comment (felix642) [reply src/lu_decomposition.c:15]: This is not something that we usually do.
I did implement this logic in the lanczos algorithm, but most algorithms are not using the last parameter as a return parameter. (Take a look, for example, at gradient descent or genetic). 

I could change the order of my parameters for this method if you think this is something that we should do.

**Pull request merged**